### PR TITLE
No longer parse request cookies, but ensure they are still redacted in the `Cookie` header string

### PR DESCRIPTION
**Pull request by stevejgordon**
We've identified that sending transactions with cookies parsed to `Request.Cookies` can result in failures to index those transactions when any of the cookies include periods (which is common for ASP.NET core by default). 

This PR removes the extraction from the header onto `Request.Cookies`. Instead, we now modify the `Cookie` header value to redact the cookie items based on the configured list of field names to sanitize. For .NET targets, we optimize this to avoid allocations as much as possible. Unfortunately, for the .NET Framework, we can't be as efficient.

### CODE REVIEW

File: src/Elastic.Apm/Api/Request.cs [hunk 1]
```diff
@@ -27,11 +27,6 @@ public class Request
 		/// </summary>
 		public Dictionary<string, string> Headers { get; set; }
 
-		/// <summary>
```
Comment (Mpdreamz) [inline src/Elastic.Apm/Api/Request.cs:30]: Breaking?

File: src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+		var redactedCookieHeader = string.Empty;
+
+#if NETFRAMEWORK
+		// The use of `Span<T>` in NETFX can cause runtime assembly loading issues due to our friend,
+		// binding redirects. For now, we take a slightly less allocation efficient route here, rather
+		// than risk introducing runtime issues for consumers. Technically, this should be "fixed" in
+		// NET472+, but during testing I surprisingly still reproduced an exception. Elastic.APM depends on
+		// `System.Diagnostics.DiagnosticSource 5.0.0` which itself depends on `System.Runtime.CompilerServices.Unsafe`
+		// which is where the exception occurs. 5.0.0 is marked as deprecated so we could look to prefer
```
Comment (Mpdreamz) [inline src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs:49]: Can you create an issue for this? Would love to bump SDD to 6.0.0 if we can :)

File: src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+		// a new version but we have special handling for the ElasticApmAgentStartupHook
+		// zip file version. For now, we decided not to mess with this as it's hard to test all scenarios.
+
+		var sb = new StringBuilder(cookieHeaderValue.Length);
```
Comment (Mpdreamz) [inline src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs:53]: Nit but can we split this into 2 implementation methods?

**Review by Mpdreamz — CHANGES_REQUESTED** [review]
Few nitpicks mostly LGTM :)

File: src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs (hunk 3 above)
Comment (stevejgordon) [reply src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs:53]: Yeah, makes sense.

File: src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs (hunk 2 above)
Comment (stevejgordon) [reply src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs:49]: https://github.com/elastic/apm-agent-dotnet/issues/2445

File: src/Elastic.Apm/Api/Request.cs (hunk 1 above)
Comment (stevejgordon) [reply src/Elastic.Apm/Api/Request.cs:30]: Yep, will document this in our release notes.

File: src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs (hunk 3 above)
Comment (stevejgordon) [reply src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs:47]: I've rearranged the compiler pre-processor directives to create two versions of the `Redact` method: one for the .NET Framework and one for everywhere else.

**Review by Mpdreamz — APPROVED** [review]

**Pull request merged**